### DISCUSSION TITLE
Can't permanently connect or transmit data from peripheral to central

### DISCUSSION THREAD

**Discussion by celiafb** [hbldh/bleak#955] [Q&A]
* bleak version: 14.3
* Python version: 3.10.5
* Operating System: macOS
* BlueZ version (`bluetoothctl -v`) in case of Linux: unsure

### Description

Using the example service_explorer.py does show the readable characteristics:

```
INFO:__main__:Connected: True
INFO:__main__:[Service] a3e0539e-3cf8-867e-2b7b-1ee451ec384b (Handle: 12): Unknown
INFO:__main__:	[Characteristic] 6e400002-b5a3-f393-e0a9-e50e24dcca9e (Handle: 13): Nordic UART RX (read,notify), Value: b'2956 2956 0 2\x00\x00\x00\x00\x00\x00'
INFO:__main__:		[Descriptor] 00002902-0000-1000-8000-00805f9b34fb (Handle: 15): Client Characteristic Configuration) | Value: b''
INFO:__main__:	[Characteristic] 6e400003-b5a3-f393-e0a9-e50e24dcca9e (Handle: 16): Nordic UART TX (write-without-response,write), Value: None

```

This is how I'm setting up my service/characteristics on the peripheral side:
```
// create service
BLEService ADCService = BLEService("A3E0539E-3CF8-867E-2B7B-1EE451EC384B");

// create characteristics
BLECharCharacteristic readCharacteristic = BLECharCharacteristic("6E400003-B5A3-F393-E0A9-E50E24DCCA9E", BLEWriteWithoutResponse | BLEWrite);
BLECharacteristic writeCharacteristic = BLECharacteristic("6E400002-B5A3-F393-E0A9-E50E24DCCA9E", BLERead | BLENotify | BLEBroadcast,19);
```

and on the central side:
```
device = "A3E0539E-3CF8-867E-2B7B-1EE451EC384B"
write_characteristic = "6e400003-b5a3-f393-e0a9-e50e24dcca9e"
read_characteristic = "6e400002-b5a3-f393-e0a9-e50e24dcca9e"
```

... and trying to read centrally using:

```
await client.start_notify(read_characteristic, handle_rx)
```

My custom script allows my central device (Mac) to originally establish a connection with the peripheral (nRF52DK), but it gets disconnected in about 2 seconds, after the **await** line above fails to run successfully. 
Using the example uart_service.py also yields a momentary connection and disconnection within a couple seconds.

### What I Did

Someone on the forum suggested that the Descriptor having a null value could be the issue. I'm very new to BLE, and I'm not sure how to set up the Descriptor on the code that I'm flashing to my peripheral, so right now I don't have any code related to that.
I did try to connect to my peripheral using the nRF Connect app on my iPhone. Doing this, I'm able to fully connect to my peripheral and continuously get the data sent over! I just want to replicate this on my computer using my bleak script.

Any advice would be greatly appreciated!

**Comment by dlech** [comment]
The Android Bluetooth stack seems to be a bit more forgiving of misbehaving devices. Logging Bluetooth packets (see troubleshooting page in Bleak docs) is generally helpful in diagnosing these types of issues.
